src/emulator/modules/SceGxm/src/SceGxm.cpp

```diff
@@ -278,13 +297,37 @@ EXPORT(int, sceGxmBeginScene, SceGxmContext *context, unsigned int flags, const
     assert(colorSurface != nullptr);
     assert(depthStencil != nullptr);
 
+    if (context->isInScene){
+        return SCE_GXM_ERROR_WITHIN_SCENE;
+    }
+    if (depthStencil == nullptr && colorSurface == nullptr){
+        return SCE_GXM_ERROR_INVALID_VALUE;
+    }
+    
     // TODO This may not be right.
     context->fragment_ring_buffer_used = 0;
     context->vertex_ring_buffer_used = 0;
     context->color_surface = *colorSurface;
-
+    
+    context->isInScene = true;
+    
     glBindFramebuffer(GL_FRAMEBUFFER, renderTarget->framebuffer[0]);
 
+    // Re-load GL machine settings for multiple contexts support
+    switch (context->cull_mode){
```

Comment:
Is this needed by anything?

Reply:
We need this to handle multiple contexts properly since several commercial games use multiple direct contexts (eg. Duke Nukem 3D).